src/components/api/json-rpc/che-json-rpc-master-api.ts

```diff
@@ -126,8 +134,13 @@ export class CheJsonRpcMasterApi {
   }
 
   reconnect(entrypoint: string): void {
-    this.$log.warn('WebSocket connection is closed.');
-    if (this.reconnectionAttemptNumber === this.maxReconnectionAttempts) {
+    if (this.reconnectionAttemptNumber < this.maxReconnectionAttempts) {
+      this.$log.warn('WebSocket connection is closed.');
+    } else {
+      this.globalWarningBannerService.addMessage(`WebSocket connections "${entrypoint}" are failing due to network restrictions.
+       ${this.branding.getName()} workspaces may not be usable. Please refer to the
```

Comment:
Is it going to be `CodeReady Workspaces workspaces may not be usable` for CRW?
https://github.com/redhat-developer/codeready-workspaces/blob/master/assembly/branding/product.json.template#L3
If yes - let's consider reformulating this.

Reply:
As for me I like existing one.

Reply:
done